packages/styles/src/tokens/_post-theme.scss

```
@@ -3,6 +3,7 @@
 @use './core';
 @use './../mixins/tokens';
 
-:root {
+:root,
+[data-color-scheme] {
```

If we want to be totally independent of the layers' order we would need to add the palette selector here as well: 
```suggestion
:root,
[data-color-scheme],
.palette {
```

---

This would mean we'd end up redeclaring tokens for any layer-selector (`.palette-default` or `[data-color-scheme]`). I think it will be hard to track where what has to be redeclared and why. This is one of the bigger unsolved issues at the moment where I see issues in the future.

In an ideal world, layers would not depend on any selectors at all, only declare on the root. Obviously that's not what we're doing right now and might not even be possible right now, but that would be the only convenient way to ensure layer order independence.

---

@alizedebray You're right, but since the semantic palette tokenset manly exists to make figma designers lives easier, by merging different variant tokens to one component token, so they have to create only one component with N variants, it makes perfect sence to me to implement this tokenset right before we link it to the component tokens.

Generally I'm with Philipp here, but if you two decide it's better to add the selector later, I won't have anything against it.

---

As I’ve mentioned before, my preferred solution would be to reorder the layers to align with CSS inheritance.
However, if we decide to keep the current order and go with the approach where layer order does not matter, we just need to be consistent and ensure the solution works regardless of the layer order. For maintainability, it’s important that selectors aren’t added arbitrarily.

---

**Resolution:** go with the inconsistency for now and investigate further in another ticket.

---

> As I’ve mentioned before, my preferred solution would be to reorder the layers to align with CSS inheritance. 

I can now invoke daoisme 😜